default/python/AI/AIstate.py

```diff
@@ -66,10 +66,7 @@ def __init__(self, aggression):
         empire = fo.getEmpire()
         self.empireID = empire.empireID
         homeworld = universe.getPlanet(empire.capitalID)
-        if homeworld:
-            self.__origin_home_system_id = homeworld.systemID
-        else:
-            self.__origin_home_system_id = INVALID_ID
+        self.__origin_home_system_id = homeworld and homeworld.systemID or INVALID_ID
```

Comment:
this does not work correctly if the homeworld.systemID is 0

but if you want a one-liner I suppose the  = if else
kind of structure could work.